oauth2client/contrib/sql_alchemy.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""OAuth 2.0 utilities for SQLAlchemy.
+
+Utilities for using OAuth 2.0 in conjunction with SQLAlchemy.
+Heavily inspired by equivalent Django module.
+
+Configuration
+=============
+
+In order to use this storage, you'll need to create table
+with :class:`oauth2client.contrib.sql_alchemy.CredentialsType` column.
+It is best used as relationship to some kind of user table.
+
+
+.. code-block:: python
+
+    from oauth2client.contrib.sql_alchemy import CredentialsType
+    from sqlalchemy import Column, ForeignKey, Integer
+    from sqlalchemy.ext.declarative import declarative_base
+    from sqlalchemy.orm import relationship
+
+
+    Base = declarative_base()
+
+
+    class Credentials(Base):
+        __tablename__ = 'credentials'
+
+        user_id = Column(Integer, ForeignKey('user.id'))
+        credentials = Column(CredentialsType)
+
+
+    class User(Base):
+        id = Column(Integer, primary_key=True)
+        # bunch of other columns
+        credentials = relationship('Credentials')
+
+
+Usage
+=====
+
+With tables ready, you are now able to store credentials in database.
+
+.. code-block:: python
+
+    from oauth2client.client import OAuth2Credentials
+    from oauth2client.contrib.sql_alchemy import Storage
+    from sqlalchemy.orm import Session
+
+    session = Session()
+    user = session.query(User).first()
+    storage = Storage(
+        session=session,
+        model_class=Credentials,
+        key_name='user_id',
+        key_value=user.id,
+        property_name='credentials',
+    )
+
+    # save
+    credentials = OAuth2Credentials(...)  # usually obtained different way
+    storage.locked_put(credentials)
+
+    # retrieve
+    credentials = storage.locked_get()
+
+    # delete
+    storage.locked_delete()
+
+"""
+
+from oauth2client.client import Storage as BaseStorage
+from sqlalchemy.types import PickleType
+
+
+class CredentialsType(PickleType):
+    """Type representing credentials.
+
+    Alias for :class:`sqlalchemy.types.PickleType`.
+    """
+
+
+class FlowType(PickleType):
```

Comment:
Yeah, I'm not using it anywhere in my code, but since it's present in other modules I've included it as well.